pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -259,6 +259,10 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 	config.RegistrationAPIServiceCABundle = encodedCaBundle
 	config.WorkAPIServiceCABundle = encodedCaBundle
 
+	if config.ImagePullSecret, err = n.getImagePullSecret(ctx); err != nil {
```

Comment:
why we need this step?

Reply:
will not add the pullsecretName to the sa if the imagepullSecret does not exist.
https://github.com/open-cluster-management-io/ocm/pull/397/files#diff-9f7acaff5c7d5c32d15173a975e8bed87d52c8add8e4bf8b4783854741c4cf39R6

if the imagepullSecret is not created but the name is fixed in the sa, a warning event will be output in the ns. The warning event can cause test case failure in openshift-conformance test suite.
There are 2 options here to avoid the warning event:
1. add a empty pullsecret in the operator namespace.
2. remove the pullsecret name from the sa if the pullsecret does not exist. 

I chose option 2 in this PR. 

the ApplyDirectly in libary-go does not update the imagepullsecret filed in the sa,  I also create a new func ApplyServiceAccount to apply sa in the PR.

Reply:
moved the imagepullsecret to deployment.